src/strategies/spidStrategy.ts

```diff
@@ -2,102 +2,81 @@
  * Builds and configure a Passport strategy to authenticate the proxy to the
  * different SPID IDPs.
  */
-
+// tslint:disable: no-object-mutation
+import { NonEmptyArray } from "fp-ts/lib/NonEmptyArray";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
 import * as SpidStrategy from "spid-passport";
+import { IDPEntityDescriptor } from "../types/IDPEntityDescriptor";
 import { SpidUser } from "../types/user";
+import { fetchIdpMetadata, parseIdpMetadata } from "../utils/idpLoader";
 import { log } from "../utils/logger";
 
-const spidStrategy = (
+/**
+ * Map provided idpMetadata in an object with idp key whitelisted in ipdIds.
+ * Mapping is based on entityID property
+ */
+export const mapIpdMetadata = (
+  idpMetadata: ReadonlyArray<IDPEntityDescriptor>,
+  idpIds: { [key: string]: string | undefined }
+) => {
```

Comment:
remove the curly braces + `return`:

```suggestion
) => idpMetadata.reduce(
```

Reply:
Fixed in a9d3aeb